animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,145 @@
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useFloatList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useFloatList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllSingle(self, object, identifier, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        indices = list(range(0, len(object.data.vertices)))
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllList(self, object, identifier, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        totalVertices = len(object.data.vertices)    
+        for i, weight in enumerate(weights):
+            if i >= totalVertices: return 
```

Comment:
Just do a `for i in range(verticesCount):` to avoid this line.

Reply:
The code below gives the error when _weights_ is equal not to _verticesCount_
```
verticesCount = len(object.data.vertices)    

for i in range(verticesCount):
    vertexGroup.add([i], weights[i], "REPLACE")
```
so I have to add another line for error handling, some thing like this:

```
verticesCount = len(object.data.vertices)    
if len(weights) < verticesCount: return object

for i in range(verticesCount):
    vertexGroup.add([i], weights[i], "REPLACE")
```
Is this correct?

Reply:
This is what I had in mind:
```python
weights = VirtualDoubleList.create(weights, 0)
for i in range(len(object.data.vertices)):
    vertexGroup.add([i], weights[i], "REPLACE")
```

Reply:
1) This code works fine when `len(object.data.vertices) = len(weights)`, however when `len(object.data.vertices) > len(weights)` then it gives an error and broke the tree. So, we have to raise an exception,
```
if len(weights) < len(object.data.vertices):
    self.raiseErrorMessage("Invalid Weights.")
weights = VirtualDoubleList.create(weights, 0)
for i in range(len(object.data.vertices)):
    vertexGroup.add([i], weights[i], "REPLACE")
```

2) The second thing is _VirtualDoubleList_ gives an error:
![Screenshot from 2019-11-08 22-49-54](https://user-images.githubusercontent.com/30294746/68497540-c7d34980-027a-11ea-8ffe-40137eef5bba.png)

Reply:
Can you try the `.get(index)` method of the virtual list?

Reply:
The code below gives an error
```
weights = VirtualDoubleList.create(weights, 0) 
print(weights.get(0))
```
and the error is:
_AttributeError: 'animation_nodes.data_structures.virtual_list.virtu' object has no attribute 'get'_

Reply:
Ok, it appears I misunderstood things, sorry. You will need to materialize the list before using it. So your code should be something like this:
```python
weights = VirtualDoubleList.create(weights, 0).materialize(verticesCount)
for i, weight in enumerate(weights):
    vertexGroup.add([i], weight, "REPLACE")
```

Reply:
Thanks, It worked.